src/adapters/pubgears.js

```diff
@@ -0,0 +1,194 @@
+
+    bids.forEach(function(bid) {
+
+      var name = getSlotFromBidParam(bid);
+      pendingSlots[ name ] = bid;
+    });
+
+    proxy = proxy || getScript(SCRIPT_ID) || makeScript(slots, publisher, SCRIPT_ID, TAG_URL);
+    if(!initialized)
+        registerEventListeners(proxy);
+    initialized = true;
+  }
+
+  function loadScript(script) {
+
+    var anchor = (function(scripts) {
+      return scripts[ scripts.length - 1 ];
+    })(d.getElementsByTagName(SCRIPT));
+
+    return anchor.parentNode.insertBefore(script, anchor);
+  }
+
+  function getSlotFromBidParam(bid) {
+
+    var size = getSize(bid);
+    var params = bid[PARAMS];
+    var slotName = params[PUB_ZONE_PARAM];
+    return [ slotName, size ].join('@');
+  }
+
+  function getSlotFromResource(resource) {
+
+    var size = resource[SIZE];
+    var key = [ resource[PUB_ZONE],  size ].join('@');
+    return key;
+  }
+  
+  function getSize(bid) {
+
+    var sizes = bid[SIZES];
+    var size = Array.isArray(sizes[0]) ? sizes[0] : sizes;
+    return size.join('x');
+  }
+
+  function makeScript(slots, publisher, id, url) {
+
+    var script = d.createElement(SCRIPT);
+    script.src = url;
+    script.id = id ;
+    script.setAttribute(ATTRIBUTE_PREFIX + SLOT_LIST_ATTRIBUTE, slots.join(' '));
+    script.setAttribute(ATTRIBUTE_PREFIX + FLAG_ATTRIBUTE, 'true');
+    script.setAttribute(ATTRIBUTE_PREFIX + PUBLISHER_ATTRIBUTE, publisher);
+
+    return loadScript(script);
+  }
+
+  function getScript(id) {
+
+    return d.getElementById(id);
+  }
+
+  function registerEventListeners(script) {
+
+    script.addEventListener(BID_RECEIVED_EVENT_NAME, onBid, true);
+    script.addEventListener(SLOT_READY_EVENT_NAME, onComplete, true);
+  }
+
+  function onBid(event) {
+
+    var data = event[DETAIL];
+    var slotKey = getSlotFromResource(data[RESOURCE]);
+    var bidRequest = pendingSlots[slotKey];
+    var adUnitCode = bidRequest[PLACEMENT_CODE];
+    var bid = null;
+
+    if(bidRequest) {
+      bid = buildResponse(data, bidRequest);
+      bidmanager.addBidResponse(adUnitCode, bid);
+      utils.logMessage('adding bid respoonse to "' + adUnitCode + '" for bid request "' + bidRequest[BID_ID] + '"');
+    }else {
+      utils.logError('Cannot get placement id for slot "' + slotKey + '"');
+    }
+  }
+
+  function buildResponse(eventData, bidRequest) {
+
+    var resource = eventData[RESOURCE];
+    var dims = resource[SIZE].split('x');
+    var price = Number(eventData[GROSS_PRICE]);
+    var status = isNaN(price) || price < 0 ? 2 : 1;
+
+    var response = bidfactory.createBid(status, bidRequest);
+    response[BIDDER_CODE_RESPONSE_KEY] = BIDDER_CODE;
+
+    if(status !== 1)
+      return response;
+
+    response[AD] = getCreative(resource);
+
+    response[CPM] = price / 1e3 ;
+    response[WIDTH] = dims[0];
+    response[HEIGHT] = dims[1];
+    return response;
+  }
+
+  function getCreative(resource) {
+
+    var bookends = '%%';
```

Comment:
You can use `utils.replaceTokenInString` for this functionality or a string template.